### ISSUE TITLE
Azure Servicebus: Sender TX ERROR on Timeout changed in version 3.x?

### ISSUE THREAD

**Issue by gjr-qm** [noodlefrenzy/node-amqp10#230]
Since i updated to version 3 (on 3.12 right now), my logs show errors that i filtered out in version 2.x.

Here is the error:
```
TX ERROR:  AmqpProtocolError: undefined:The link '...' is force detached by the broker due to errors occurred in publisher(link...). Detach origin: AmqpMessagePublisher.IdleTimerExpired: Idle timeout: 00:10:00. TrackingId:...,TimeStamp:4/1/2016 11:50:18 AM
```

My code that used to work in version 2.x:
```
sender.on('errorReceived', function(tx_err) {
    if (tx_err.condition && (tx_err.condition == 'amqp:link:detach-forced')) {
        // Not a real error, Azure timeout
        amqpTopicClient.disconnect().then(function () {
            // clean up connection cache
            logger.info(`Disconnected (queuename)`);
        });
    }
    else {
        // Real error
       logger.error(queuename + '===> TX ERROR: ', tx_err);                    
    }
})
```

The condition on the tx_err is undefined as of version 3.x

What is the best way to handle these timeout messages in the new version?

**Comment by mbroadst** [comment]
@gjr-qm hrm that's really strange, can you possibly print out debug information around the time of the force detach? It looks like the condition isn't being properly caught.

You would just run your code with `DEBUG=amqp*` before it (or defined in the env vars on windows iirc)

**Comment by mbroadst** [comment]
@gjr-qm ah I see the problem right here: https://github.com/noodlefrenzy/node-amqp10/blob/6e20ff9a123b2f78f0fe24401dfc7a3c6a9b3d7d/lib/utilities.js#L265. I'll put together a quick patch and test and we'll get 3.1.3 out the door for you

**Comment by gjr-qm** [comment]
Cool! Thanks for your quick response (as always). If you need more info from me, i'll be happy to help out.

**Comment by mbroadst** [comment]
@gjr-qm should be fixed in 3.1.3, cheers

**Comment by gjr-qm** [comment]
Perfect, it works as before! Thanks again, cheers.
